app/src/main/java/com/onewelcome/showcaseapp/feature/userderegistration/UserDeregistrationScreen.kt

```diff
@@ -0,0 +1,164 @@
+package com.onewelcome.showcaseapp.feature.userderegistration
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.onFailure
+import com.github.michaelbull.result.onSuccess
+import com.onegini.mobile.sdk.android.handlers.error.OneginiError
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onewelcome.core.components.SdkFeatureScreen
+import com.onewelcome.core.components.ShowcaseDropdownMenu
+import com.onewelcome.core.components.ShowcaseFeatureDescription
+import com.onewelcome.core.components.ShowcaseStatusCard
+import com.onewelcome.core.theme.Dimensions
+import com.onewelcome.core.theme.separateItemsWithComa
+import com.onewelcome.core.util.Constants
+import com.onewelcome.showcaseapp.R
+import com.onewelcome.showcaseapp.feature.userderegistration.UserDeregistrationViewModel.State
+import com.onewelcome.showcaseapp.feature.userderegistration.UserDeregistrationViewModel.UiEvent
+
+@Composable
+fun UserDeregistrationScreen(
+  navController: NavController,
+  viewModel: UserDeregistrationViewModel = hiltViewModel()
+) {
+  UserDeregistrationScreenContent(
+    viewModel.uiState,
+    { navController.popBackStack() },
+    { viewModel.onEvent(it) }
+  )
+}
+
+@Composable
+private fun UserDeregistrationScreenContent(
+  uiState: State,
+  onNavigateBack: () -> Unit,
+  onEvent: (UiEvent) -> Unit
+) {
+  LaunchedEffect(Unit) {
+    onEvent.invoke(UiEvent.LoadInitialData)
+  }
+  SdkFeatureScreen(
+    title = stringResource(R.string.section_title_user_deregistration),
+    onNavigateBack = onNavigateBack,
+    description = {
+      ShowcaseFeatureDescription(
+        stringResource(R.string.user_deregistration_description),
+        Constants.DOCUMENTATION_USER_DEREGISTRATION
+      )
+    },
+    settings = { SettingsSection(uiState, onEvent) },
+    result = uiState.result?.let { { DeregistrationResult(it) } },
+    action = { DeregisterUserButton(uiState, onEvent) }
+  )
+}
+
+@Composable
+private fun SettingsSection(uiState: State, onEvent: (UiEvent) -> Unit) {
+  Column(verticalArrangement = Arrangement.spacedBy(Dimensions.verticalSpacing)) {
+    SdkInitializationSection(uiState.isSdkInitialized)
+    RegisteredUserProfilesSection(uiState.registeredUserProfiles)
```

Comment:
I would not divide `RegisteredUserProfilesSection` and `UserProfileSelectionSection` into two. We're duplicating same info. 

I think better approach would be to show a list, similar to what we have for Identity Providers in Browser Registration. Such a radio button list is more clear, descriptive and aligns with material design principles. 

If there's no user profiles, we could either show what we have already - empty RegisteredUserProfilesSection OR we could have an empty radio button list with text saying that there's no user profiles to select.